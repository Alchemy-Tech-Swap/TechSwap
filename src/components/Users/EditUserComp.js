import React from 'react';
import './Users.css';

export default function EditUserComp({ label, name, onChange, slack_id, linkedin_url, onSubmit }) {
  return (
    <div>
      <div>EDIT</div>
      <div>{label || 'User Info'}</div>
      <form onSubmit={(e) => onSubmit(e)}>
        <label htmlFor="name">Name: </label>
        <input id="name" name="name" type="text" value={name} onChange={(e) => onChange(e)} />
        <br />
        <label htmlFor="slack_id">Slack: </label>
        <input
          id="slack_id"
          name="slack_id"
          type="text"
          value={slack_id}
          onChange={(e) => onChange(e)}
        />
        <br />
        <label htmlFor="linkedin_url">LinkedIn: </label>
        <input
          id="linkedin_url"
          name="linkedin_url"
          type="text"
          value={linkedin_url}
          onChange={(e) => onChange(e)}
        />
        <button>Submit</button>
      </form>
    </div>
  );
}
