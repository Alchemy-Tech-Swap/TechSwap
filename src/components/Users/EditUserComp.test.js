import EditUserComp from './Users';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test('Should render Edit User Component', async () => {
  const { container } = render(
    <MemoryRouter>
      <EditUserComp
        user={{
          label: 'Edit User Info',
          name: 'rewqrewq',
          slack_id: 'rewqrewq',
          linkedin_url: 'rewqrewq',
          onChange: 'ƒ onChange() {}',
          onSubmit: 'ƒ onSubmit() {}',
        }}
      />
    </MemoryRouter>
  );

  await screen.findByText('Profile');

  expect(container).toMatchSnapshot();
});
