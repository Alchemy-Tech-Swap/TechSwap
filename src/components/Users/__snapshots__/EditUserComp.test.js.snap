// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Edit User Component 1`] = `
<div>
  <div>
    <h1>
      Profile
    </h1>
    <div>
      Name: 
      rewqrewq
    </div>
    <div>
      Email: 
    </div>
    <div>
      LinkedIn: 
      rewqrewq
    </div>
    <div>
      Slack: 
      rewqrewq
    </div>
  </div>
</div>
`;
